File: StudyAsist-Android/app/src/main/java/com/studyasist/data/local/entity/AssessmentQuestion.kt
package com.studyasist.data.local.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "assessment_questions",
    foreignKeys = [
        ForeignKey(
            entity = Assessment::class,
            parentColumns = ["id"],
            childColumns = ["assessmentId"],
            onDelete = ForeignKey.CASCADE
        ),
        ForeignKey(
            entity = QA::class,
            parentColumns = ["id"],
            childColumns = ["qaId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index(value = ["assessmentId"]), Index(value = ["qaId"])]
)
data class AssessmentQuestion(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val assessmentId: Long,
    val qaId: Long,
    val weight: Float = 1.0f,
    val sequence: Int
)

File: StudyAsist-iOS/StudyAsist/Models/AssessmentQuestion.swift
//
//  AssessmentQuestion.swift
//  StudyAsist
//

import Foundation
import SwiftData

@Model
final class AssessmentQuestion {
    @Attribute(.unique) var id: Int64
    var assessmentId: Int64
    var qaId: Int64
    var weight: Float
    var sequence: Int
    
    init(
        id: Int64 = 0,
        assessmentId: Int64,
        qaId: Int64,
        weight: Float = 1.0,
        sequence: Int
    ) {
        self.id = id
        self.assessmentId = assessmentId
        self.qaId = qaId
        self.weight = weight
        self.sequence = sequence
    }
}
